### ISSUE TITLE
Install in Ubuntu 14

### ISSUE THREAD

**Issue by pdelamora** [TRIQS/triqs#187]
Dear All,
I could not install TRIQS into a Fedora Linux so I moved to Ubuntu
*.*.*.*
I installed 
Ubuntu 14.04.1 LTS
I installed the required libraries
*.*.*.*
In
TRIQS/build
I put
cmake ../src/
and I get
*-*-*-*
You are using the gcc compiler but your compiler is too old :
   TRIQS requires version >= 4.8.1 while you have
*-*-*-*
I ran
apt-get upgrade gcc
and I get
gcc ya está en su versión más reciente
=> gcc is in its most recent version
but I still get the same response from
cmake ../src/
*.*.*.*
So I upgraded to the latest
Ubuntu 14.10
with no improvement in this issue

Pablo de la Mora

**Comment by tayral** [comment]
|Dear Pedro,
What is your g++ version?
(>g++ --version)
Ubuntu 14.04 should ship with g++ 4.8 if I am not mistaken


Should you nonetheless want to install g++ 4.9 :
sudo add-apt-repository ppa:ubuntu-toolchain-r/test
sudo apt-get update
sudo apt-get install g++-4.9

and then specify the compiler at the 'cmake' stage:
CXX=g++-4.9 cmake -DCMAKE_INSTALL_PREFIX ...


|

**Comment by pdelamora** [comment]
Dear Thomas,

        Thank you very much.

        By the way, I am Pablo (Paul) and not Pedro (Peter), ha, ha.

        *.*.*.*

Yes, I assumed that Ubuntu 14.04 would have the latest

g++

*.*.*.*

I did what you suggested;

sudo apt-get install g++-4.9

=>

g++-4.9 ya está en su versión más reciente=> is in the most recent version


I tried, as you suggested


CXX=g++-4.8 cmake ../src/

=>

You are using the gcc compiler but your compiler is too old :
   TRIQS requires version >= 4.8.1 while you have

  ************************** FATAL ERROR ************************************

*.*.*.*

It is interesting that it requires 4.8.1 and it says

"while you have      "

but does not say which gcc I have!!!

    Saludos


                Pablo

________________________________
De: Thomas Ayral <notifications@github.com>
Enviado: jueves, 5 de febrero de 2015 03:46 p. m.
Para: TRIQS/triqs
Cc: delamora
Asunto: Re: [triqs] Install in Ubuntu 14 (#187)

|Dear Pedro,
What is your g++ version?
(>g++ --version)
Ubuntu 14.04 should ship with g++ 4.8 if I am not mistaken


Should you nonetheless want to install g++ 4.9 :
sudo add-apt-repository ppa:ubuntu-toolchain-r/test
sudo apt-get update
sudo apt-get install g++-4.9

and then specify the compiler at the 'cmake' stage:
CXX=g++-4.9 cmake -DCMAKE_INSTALL_PREFIX ...


|

Le 05/02/2015 22:16, pdelamora a écrit :
>
> Dear All,
> I could not install TRIQS into a Fedora Linux so I moved to Ubuntu
> /./././
> I installed
> Ubuntu 14.04.1 LTS
> I installed the required libraries
> /./././
> In
> TRIQS/build
> I put
> cmake ../src/
> and I get
> /-/-/-/
> You are using the gcc compiler but your compiler is too old :
> TRIQS requires version >= 4.8.1 while you have
> /-/-/-/
> I ran
> apt-get upgrade gcc
> and I get
> gcc ya está en su versión más reciente
> => gcc is in its most recent version
> but I still get the same response from
> cmake ../src/
> /./././
> So I upgraded to the latest
> Ubuntu 14.10
> with no improvement in this issue
>
> Pablo de la Mora
>
> -
> Reply to this email directly or view it on GitHub
> <https://github.com/TRIQS/triqs/issues/187>.
>


-

**Comment by pseth** [comment]
Pablo, did you try CXX=g++-4.8 cmake ../src/ or CXX=g++-4.9 cmake ../src/ ? Could you please try the latter?

BTW executing "gcc --version" will give you your default gcc version.

**Comment by pdelamora** [comment]
Yes, I tried both, same result.

________________________________
De: Priyanka Seth <notifications@github.com>
Enviado: jueves, 5 de febrero de 2015 04:22 p. m.
Para: TRIQS/triqs
Cc: delamora
Asunto: Re: [triqs] Install in Ubuntu 14 (#187)


Pablo, did you try CXX=g++-4.8 cmake ../src/ or CXX=g++-4.9 cmake ../src/ ? Could you please try the latter?

BTW executing "gcc --version" will give you your default gcc version.

-

**Comment by mferrero** [comment]
You need to completely erase your build directory before typing ```CXX=g++-4.9 cmake ../src/```. Otherwise the cmake cache file will remember the old 4.8 compiler.

**Comment by parcollet** [comment]
Still, it should work out of the box with the default on 14.04, which is gcc 4.8.2

**Comment by parcollet** [comment]
What is g++ --version ?

**Comment by pdelamora** [comment]
Dear Michel,

        You were right, I erased the


TRIQS


directory and reloaded it.

Then I wrote


cmake ../src/


and it ran without problems


        Also thank all for their help!


                    Pablo

*.*.*.*.*.*

De: Michel Ferrero <notifications@github.com>



You need to completely erase your build directory before typing CXX=g++-4.9 cmake ../src/. Otherwise the cmake cache file will remember the old 4.8 compiler.

-

**Comment by pdelamora** [comment]
Thank you for all the help
*************
As mentioned above I had problems with the gcc, but as Michel Ferrero commented I need to erase the build directory.
After this I had no problems.
So I decided to start from the beginning, that is;
*************************
I restarted installing Ubuntu 14.04.1LTS
I installed the packages and libboost
after this I went all the way to 
*     make
*     make install

All this ran smoothly

*********************
In conclusion it seems that the installation in Linux Fedora is not that simple and the change to Ubuntu proved to be a good alternative.
* I thank all of you for your help, in special to Markus Aichhorn for all his answers and in particular for suggesting to move to Ubuntu 
* Now I have to solve some of the problems on how to work with the WIEN2k package in this linux version
